src/transaction_processor.py

```diff
@@ -0,0 +1,182 @@
+from typing import List, Tuple
+from src.helpers.blockchain_data import BlockchainData
+from src.helpers.database import Database
+from src.imbalances_script import RawTokenImbalances
+from src.price_providers.price_feed import PriceFeed
+from src.helpers.helper_functions import read_sql_file
+from src.helpers.config import CHAIN_SLEEP_TIME, logger
+import time
+
+
+class TransactionProcessor:
+    """Class processes transactions for the slippage project."""
+
+    def __init__(self, blockchain_data: BlockchainData, db: Database, chain_name: str):
+        self.blockchain_data = blockchain_data
+        self.db = db
+        self.chain_name = chain_name
+        self.imbalances = RawTokenImbalances(self.blockchain_data.web3, self.chain_name)
+        self.price_providers = PriceFeed()
+        self.log_message: List[str] = []
+
+    def get_start_block(self) -> int:
+        """
+        Retrieve the most recent block already present in raw_token_imbalances table,
+        delete entries for that block, and return this block number as start_block.
+        If no entries are present, fallback to get_finalized_block_number().
+        """
+        try:
+            # Query for the maximum block number
+            query_max_block = read_sql_file("src/sql/select_max_block.sql")
+            result = self.db.execute_query(
+                query_max_block, {"chain_name": self.chain_name}
+            )
+            row = result.fetchone()
+            max_block = row[0] if row is not None else None
+
+            if max_block is not None:
+                logger.debug("Fetched max block number from database: %d", max_block)
```

Comment:
Is it intended there is no matching log message for fetching the block from chain?

Reply:
Yes, and fetching the block from chain is just a one time log. Also calling logger from config file uses our single logger instance with the name "raw_token_imbalances" (can be renamed)

Reply:
I would have expected a log especially in an uncommon case. (The logger should be renamed as _raw_ token imbalances is not what the repo will be about. Or just create one logger per context.)